### ISSUE TITLE
【feature request】Adding an option to render tasks color according to defined tag colors in the all tasks view

### ISSUE THREAD

**Issue by Ravenclawer** [haydenull/logseq-plugin-agenda#303]
This is a great plugin 
The task assignment logic in 3.0 is amazing 
One small change I desire is rendering tasks color according to their tags 
One can only see the color when implementing a tag filter. However, In the all tasks view, the color is defined by page.
I'd like to know if you can implement an option to show the tag-defined color and the page defined color in the all-tasks view (when conflicting, using tag defined color).

**Comment by haydenull** [comment]
Do you want to set a color for the entire task card?

I'm afraid it's not quite appropriate. A task may have multiple tags, and having so many task cards would affect the visual effect.

**Comment by Ravenclawer** [comment]
Or put it like this: defining color at block level instead of page level

**Comment by Rocreator** [comment]
Yeah, it would be really nice. I would tell you why in my specific use case: 

I divised three ways of doing it:
- I had this tasks that I have inserted in a template  that I load each day in the journal page, so I could easy view them in agenda toggling the "Use journal Day as Schedule" option in settings. This would be the easiest but for now no color in any of the views the plugin shows if used the page tag.
- Set up a recurring event. It's not feasible because the plugin does not allow to time-block it, it's an "all day" task
- Using smartblocks plugin to create a button in the template that creates a task in the desired page but I will have to go there and schedule it (so it appears in agenda), because the schedule mechanism uses a different format than smartblock's NLP (simple journal day link [[x day]]). This is doable with only a few task but becomes huge with great workloads, even being "just" an O(n) problem. 

It would be nice if at least you could make it so the first tag that appears is the one used for the page colour system. 

Really thanks for you time in this project, it has already helped a lot 🫶

**Comment by haydenull** [comment]
I think filters should also be able to achieve a similar effect as tags, for example, creating a filterA to query all tasks with tagA, then setting a color for this filter would suffice. You can refer to this document. https://www.notion.so/haydenut/English-fd47d1843e654f5485a07ef4935ed365#7bd975fe6d07453686a22d264c228f81

**Comment by Rocreator** [comment]
Ohhh yeah I have seen filter uses.
But it will filter out the rest of the task that do not have this tags, right? How to do it it I want a bird eye view of all the week or month? Or in the "plan" section?

**Comment by haydenull** [comment]
‌‌You can create two filters, one is filterA to collect all tasks with tagA, and the other is Others to collect all tasks.

Then a task may simultaneously match two filters, and Agenda will select the color of the first filter as the color of the task.

![CleanShot 2024-05-27 at 10 25 49](https://github.com/haydenull/logseq-plugin-agenda/assets/17695989/d950b345-0685-4dfa-a843-5312d4d1722a)
![CleanShot 2024-05-27 at 10 25 07](https://github.com/haydenull/logseq-plugin-agenda/assets/17695989/36f7cbdf-ced5-4e8d-888b-2a4456486be5)
![CleanShot 2024-05-27 at 10 25 27](https://github.com/haydenull/logseq-plugin-agenda/assets/17695989/087ae816-4d04-4acd-830e-df4f992fef50)

**Comment by Ravenclawer** [comment]
here is a good example of being able to define the color for each task
let's say we have a project page named “XXXX ESSAY WRITING”
and we have to write the essay， and have a discussion with lucy and then submit the essay 
for writing，it‘s a high-priority task
for discussion，lunch with lucy is a fixed task，since there is only one lunch at a day.
for coffee, it;s a flexible task, if we haven't finish essay, we can replace it with writing essay task 
for submission, it;s a deadline. if the submission system close, we can't submit essay anymore.
![image](https://github.com/haydenull/logseq-plugin-agenda/assets/73892411/9917a41e-0212-4466-b41d-a1f49e24f39a)
So it's better to define color at block level, since not all tasks in a project is of the same type and it's good to have an option to define them in an easy and intuitive way. applying filters is complicated and easy to omit some tasks.
